index.js

```diff
@@ -138,3 +138,76 @@ newSightingForm.addEventListener('submit', function (event) {
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
 
+// EVENT DELEGATION
+// Identify which elements you care to listen for an event on
+//      'clicks'
+
+// delete and likes buttons of all the posts
+// Find the closest common parent of all these elements
+//      <ul id="animals"> == $0
+
+
+//can click on 'like' 'delete' and 'toggle update form' for each animal sighting post
+const animalsCollection = document.querySelector('ul#animals')
+
+animalsCollection.addEventListener('click', function (event) {
+
+    const li = event.target.closest('li')
+    
+    // before review
+    // if (event.target.matches('button.like-button')){
+    //     const likesPtag = event.target.previousElementSibling
+    //     const likes = parseInt(likesPtag.textContent) + 1
+    //     likesPtag.textContent = `${likes} Likes`
+    // }
+
+    //after review
+    if (event.target.matches('button.like-button')){
+        const likesPtag = li.querySelector('p.likes-display')
+        const currlikes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${currlikes} Likes`
+    }
+
+    //before review
+    // else if (event.target.matches('button.delete-button')) {
+    //     const animals = event.target.closest('li')
+    //     animals.remove()
+    // }
+    // getting error: index.js:163 Uncaught TypeError: Cannot read property 'remove' of null
+    // at HTMLUListElement.<anonymous> (index.js:163)
+    // when trying to click on 'delete' button. not sure what is wrong with the code.
+
+    //after review
+    else if(event.target.className === 'delete-button') {
+        li.remove()
+    }
+    else if(event.target.matches('button.toggle-update-form-button')) {
+        
+
+         const form = li.querySelector('form')
+
+        // if(form.style.display === 'block') {
+        //     form.style.display
+        // }
+        // else {
+        //     form.style.display = 'block'
+        // }
+
+        form.style.display = form.style.display === 'block' ? 'none' : 'block'
+
+    }
+    
+})
+
+//after review
+animalsCollection.addEventListener('submit', function(event) {
+    if (event.target.matches('form.update-form')) {
+        event.preventDefault()
+        
+        const descriptionInput = event.target[0].value
+        const li = event.target.closest('li')
+        const descriptionPtag = li.querySelector('p')
+        descriptionPtag.textContent = descriptionInput
+
+    }
+})
```

Comment:
Great work here as well. You are listening for any form submission that occurs within the animals UL element. Then you make sure to prevent the default action of the form. Then you find the p tag and update it's value based on the form data. Don't forget to change the form field value to the new description. Again, we can break it apart into 2 steps.

```
/***** BONUS DELIVERABLE part 1 *****/

animalsUL.addEventListener("submit", function(event){
    event.preventDefault()
    if (event.target.matches('.update-form')){
        handleUpdateFormSubmission(event)
    }
})

/***** BONUS DELIVERABLE part 2 *****/
function handleUpdateFormSubmission(e){
    e.preventDefault()
    const description = e.target.children[0].value
    const animalDescriptionPTag = e.target.parentElement.children[0]
    animalDescriptionPTag.innerText = description
    e.target.value = description
}
```